compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InterfaceLowering.kt

```diff
@@ -94,7 +97,7 @@ private class InterfaceLowering(val context: JvmBackendContext) : IrElementTrans
         }
     }
 
-    private fun shouldRemoveFunction(function: IrFunction): Boolean =
+    private fun mustMoveToDefaultImpls(function: IrFunction): Boolean =
         Visibilities.isPrivate(function.visibility) && function.name != clinitName ||
                 function.origin == IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER ||
```

Comment:
Please note that for @JvmDefault function  '$default' companion is not moved to DefaultImpls class (except maybe compatibility mode)

Reply:
OK, I think I understand the behaviour. In the case of @JvmDefault+compat, $default is implemented on the interface itself with DefaultImpls containing a simple delegation to that implementation.

With this PR, JVM_IR is on par with the old backend.

Could you clarify your intent with this comment? I think neither name (shouldRemove/mustMove) is true to the behaviour for FUNCTION_FOR_DEFAULT_PARAMETER.

Reply:
Seems this comment is more suitable for line 56 ``` if(function.hasJvmDefault() && !context.state.jvmDefaultMode.isCompatibility && !mustMoveToDefaultImpls(function)) continue```. Maybe flag about @JvmDefault should be passed to this funtion

Reply:
@kandersen My point was to fix a bit behaviour with  '$default' companion function, but it's could be postponed for another commit.

Reply:
I have a major revision of this entire PR in the works, I think I might have adressed this in there. We'll see once I land it :)